src/mod/l4_thruster.c

```diff
@@ -332,55 +416,117 @@ void ParseReadRequest(uint8_t* received_buffer,int len){
 	if(uint16_payload_length >5){
 
 		//if payload is more then 4 then we are reading multiple registers
-		//printf("READING MULTIPLE REGISTERS");
+
+		//sprintf(print_str, "\n Parse Request : Payload Length = %d \n",uint16_payload_length  );
+		//len_print = strlen(print_str);
+		//deb_print_pure_debug((uint8_t *)print_str, len_print);
+
+
+		//sprintf(print_str, "\n Reading Multiple Registers \n"  );
+		//len_print = strlen(print_str);
+		//deb_print_pure_debug((uint8_t *)print_str, len_print);
 
 		// Lets say that READ REQUEST TO ALL REGISTERS/ READ MORE THEN ONE REGISTER function will set last address to a value
 		// of a first address that is intended to be accessed
 		uint8_t next_register_index = LATEST_ACCESSED_REGISTER;
 		uint8_t length_of_next_register = REGISTER_LENGTH[next_register_index];
 		double multiplier = CONVERSION_DOUBLE[next_register_index];
+		//uint8_t starting_register = LATEST_ACCESSED_REGISTER; // first time
+
+		///sprintf(print_str, "\n Start RI= %d_RI_len=%d  Conversion_multiplier = %.2f \n",next_register_index,length_of_next_register,multiplier  );
+		//len_print = strlen(print_str);
+		//deb_print_pure_debug((uint8_t *)print_str, len_print);
+
+		int i = 0; // i referres to position in received_datavector, not register_index. Starting i is always 0.
+		uint32_t parse_start_time = (uint32_t)timGetSystime();
+		uint32_t current_time;
+		while( next_register_index<108 ){ // Untill last usable register !  WARNING !!! CHECK LENGTH OF ReadAllRegistersRequest !! This might trigger infitite loop
+			// It will trigger infinite loop if thruser reply with length of message that is greaater then 5 and less then enough to cover 101 registers !
+			//sprintf(print_str, "\nParse Iteration = %d length_of_next_register = %d \n", i,length_of_next_register);
+			//len_print = strlen(print_str);
+			//deb_print_pure_debug((uint8_t *)print_str, len_print);
+
+			//sprintf(print_str, "\n Next RI= %d Next_RI_len=%d  Conversion_multiplier = %.2f \n",next_register_index,length_of_next_register,multiplier  );
+			//len_print = strlen(print_str);
+			//deb_print_pure_debug((uint8_t *)print_str, len_print);
+			current_time = (uint32_t)timGetSystime();
+
+			if((current_time - parse_start_time)> 100){
```

Comment:
But still - this exit condition is a MUST !

If for some reason thruster will reply less bytes. Or some bytes arre corrupted or lost. To avoid stuck into infinite loop we add break condition based on maximum allowed proccessing time. 

It should be less then 100ms. 


I will probably formulate another issue regarding parser. But for next design iterations.

For now it should be fine